File: Programming-Basics-With-JavaScript/Homeworks-And-Labs/04.For-Loop/03-additional/02.hospital.js
function hospital(input) {

    let doctorsCount = 7;

    const period = Number(input[0]);
    let index = 1;

    let treatedPatients = 0;
    let untreatedPatients = 0;

    for (let i = 1; i <= period; i++) {
        let patientsForCurrentDay = Number(input[index]);
        index++;
        if (i % 3 === 0 && untreatedPatients > treatedPatients) {
            doctorsCount += 1;
        }

        if (patientsForCurrentDay > doctorsCount) {
            treatedPatients += doctorsCount;
            untreatedPatients += patientsForCurrentDay - doctorsCount;
        } else if (patientsForCurrentDay === doctorsCount || doctorsCount > patientsForCurrentDay) {
            treatedPatients += patientsForCurrentDay;
        }
    }

    console.log(`Treated patients: ${treatedPatients}.`);
    console.log(`Untreated patients: ${untreatedPatients}.`);
}
hospital(["3", "7", "7", "7"]);

File: Python_Basics/Homeworks-And-Labs/04.for_loop/03.additional/02.hospital.py
doctors_count = 7

period = int(input())

treated_patients = 0
untreated_patients = 0

for i in range(1, period + 1):
    patients_for_current_day = int(input())
    if i % 3 == 0 and untreated_patients > treated_patients:
        doctors_count += 1

    if patients_for_current_day > doctors_count:
        treated_patients += doctors_count
        untreated_patients += patients_for_current_day - doctors_count
    elif patients_for_current_day == doctors_count or doctors_count > patients_for_current_day:
        treated_patients += patients_for_current_day

print(f'Treated patients: {treated_patients}.')
print(f'Untreated patients: {untreated_patients}.')
